scvi/core/posteriors/posterior.py

```diff
@@ -104,7 +104,9 @@ def __init__(
                         key
                     )
                 )
-        self.gene_dataset = BioDataset(adata, getitem_tensors=self._data_and_attributes)
+        self.gene_dataset = ScviDataset(
```

Comment:
can we also remove all self.gene_dataset references to just dataset? This might be over a few files like trainers.